doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
```

Comment:
Agree with @rim-odoo : CamelCase for new api, odoo_case for old api for compatibility